cmd/containerd-shim-runc-v2/runc/container.go

```diff
@@ -301,8 +299,8 @@ func (c *Container) CgroupSet(cg interface{}) {
 
 // Process returns the process by id
 func (c *Container) Process(id string) (process.Process, error) {
-	c.mu.Lock()
-	defer c.mu.Unlock()
+	c.mu.RLock()
+	defer c.mu.RUnlock()
```

Comment:
Assuming that `c.process` is immutable, we should be able to return without locking when `id == ""`.
Are we sure it is immutable?